lib/src/query_component.dart

```diff
@@ -189,40 +190,33 @@ class QueryComponent {
     return ReportingQueryParser.parseRevenueFromJsonString(report);
   }
 
-  List<InsertionOrder> _filterInFlightInsertionOrders(
-          List<InsertionOrder> insertionOrders) =>
-      insertionOrders
-          .where((io) =>
-              io.budget.activeBudgetSegment !=
-              InsertionOrder_Budget_BudgetSegment())
-          .toList();
-
-  DateTime _getMinStartDate(
-          List<InsertionOrder> insertionOrders) =>
-      insertionOrders
-          .map((io) => Util.convertProtoDateToDateTime(
-              io.budget.activeBudgetSegment.dateRange.startDate))
-          .reduce((date, minDate) => date.isBefore(minDate) ? date : minDate);
-
   void _addSpentToInsertionOrders(List<InsertionOrder> insertionOrders,
       Multimap<String, InsertionOrderDailySpend> spendingMap) {
     for (final io in insertionOrders) {
+      // Continues if the spending map doesn't have spending data for this io
+      // at all.
+      if (!spendingMap.containsKey(io.insertionOrderId)) continue;
+
       final budgetUnit = io.budget.budgetUnit;
       final flightStart = Util.convertProtoDateToDateTime(
           io.budget.activeBudgetSegment.dateRange.startDate);
       final flightEnd = Util.convertProtoDateToDateTime(
           io.budget.activeBudgetSegment.dateRange.endDate);
 
-      final spent = spendingMap[io.insertionOrderId]
+      final activeSpent = spendingMap[io.insertionOrderId]
           .where((dailySpend) =>
               Util.isBetweenDates(dailySpend.date, flightStart, flightEnd))
           .map((dailySpend) => budgetUnit ==
                   InsertionOrder_Budget_BudgetUnit.BUDGET_UNIT_CURRENCY
               ? dailySpend.revenue
-              : dailySpend.impression)
-          .reduce((spend, sum) => Util.addStringRevenue(spend, sum));
-
-      io.spent = spent;
+              : dailySpend.impression);
+
+      // Continues if the spending map doesn't have spending data for this io
+      // within [reportStartDate, now].
+      if (activeSpent.isEmpty) continue;
```

Comment:
This case will happen when the IO has historical spending data but no active ones.